GeoLib/MinimalBoundingSphere.cpp

```diff
@@ -154,7 +154,9 @@ MinimalBoundingSphere MinimalBoundingSphere::recurseCalculation(std::vector<GeoL
             if (i>start_idx)
             {
                 GeoLib::Point* tmp = sphere_points[start_idx+i];
-                std::copy(sphere_points.begin() + start_idx, sphere_points.begin() + (start_idx + i), sphere_points.begin() + (start_idx + 1));
+                std::copy(sphere_points.begin()+static_cast<std::vector<GeoLib::Point*>::iterator::difference_type>(start_idx),
+                    sphere_points.begin()+static_cast<std::vector<GeoLib::Point*>::iterator::difference_type>(start_idx + i),
+                    sphere_points.begin()+static_cast<std::vector<GeoLib::Point*>::iterator::difference_type>(start_idx + 1));
```

Comment:
To avoid a warning about the types the std::size_t has to be static_casted to std::vector<GeoLib::Point*>::iterator::difference_type. I will use a type alias as @endJunction suggested in order to make it readable.